tests/test_db_connector.py

```diff
@@ -0,0 +1,47 @@
+from models import User
+import pytest
+from db_connector import DbConnector
+
+
+@pytest.fixture(scope='module')
+def users_list():
+    return [User(name='Henry', age=20, address='hyderabad'),
+            User(name='James', age=30, address='Bengaluru')]
+
+
+def test_Database_not_supported_wrong_database_type():
+    with pytest.raises(SystemExit) as pytest_wrapped_e:
+        DbConnector(('DB2', 'test_pg_database',
+                     'root', 'root', 'localhost'))
+        assert pytest_wrapped_e.type == SystemExit
+        assert pytest_wrapped_e.value.code == 42
+        assert pytest_wrapped_e.value.message == 'Database not supported'
+
+
+def test_database_authentication_error_wrong_username_password():
+    with pytest.raises(SystemExit) as e:
+        DbConnector(('postgres', 'test_pg_database',
```

Comment:
You should use mock, instead of connecting to actual database. Assuming these are unit tests, unit tests doesn't need to actually connect to db.

Reply:
You should test each method and look for all scenarios and use mocking/stubbing for things that have external dependencies.